luigi/task.py

```diff
@@ -47,27 +47,72 @@
 TASK_ID_TRUNCATE_PARAMS = 16
 TASK_ID_TRUNCATE_HASH = 10
 TASK_ID_INVALID_CHAR_REGEX = re.compile(r'[^A-Za-z0-9_]')
+_SAME_AS_PYTHON_MODULE = '_same_as_python_module'
 
 
-def namespace(namespace=None):
+def namespace(namespace=None, scope=''):
     """
     Call to set namespace of tasks declared after the call.
 
-    It is best practice to call this function without arguments at the end of any file it has been
-    used in. That is to ensure that subsequent tasks have the default namespace again.
+    It is often desired to call this function with the keyword argument
+    ``scope=__name__``.
+
+    The ``scope`` keyword makes so this call is only affective for task classes
+    with a matching [*]_ ``__module__``. The default value for ``scope`` is the
```

Comment:
English correction (if understand correctly):

The ``scope`` keyword makes __*it*__ so __*that*__ this call is only __*effective*__ for task classes with a matching [*]_``__module__``.